iree/compiler/Dialect/VMLA/IR/test/general_ops.mlir

```diff
@@ -92,22 +92,51 @@ func @vmla_batch_matmul(%lhs : !vmla.buffer,
 
 // -----
 
-// CHECK-LABEL: @vmla_transpose
-// CHECK-SAME: %[[SRC:[a-zA-Z0-9]+]]
-// CHECK-SAME: %[[SRC_SHAPE:[a-zA-Z0-9]+]]
-// CHECK-SAME: %[[DST:[a-zA-Z0-9]+]]
-// CHECK-SAME: %[[DST_SHAPE:[a-zA-Z0-9]+]]
-func @vmla_transpose(%src : !vmla.buffer,
-                     %src_shape : !shapex.ranked_shape<[64,32,32,10]>,
-                     %dst : !vmla.buffer,
-                     %dst_shape : !shapex.ranked_shape<[64,10,32,32]>) {
-  // CHECK:      vmla.transpose
-  // CHECK-SAME: %[[SRC]](%[[SRC_SHAPE]] : !shapex.ranked_shape<[64,32,32,10]>),
-  // CHECK-SAME: out
-  // CHECK-SAME: %[[DST]](%[[DST_SHAPE]] : !shapex.ranked_shape<[64,10,32,32]>)
-  // CHECK-SAME: {permutation = dense<[0, 3, 2, 1]> : tensor<4xi32>} : f32
-  vmla.transpose %src(%src_shape : !shapex.ranked_shape<[64,32,32,10]>),
-                 out %dst(%dst_shape : !shapex.ranked_shape<[64,10,32,32]>)
-                 {permutation = dense<[0, 3, 2, 1]> : tensor<4xi32>} : f32
+// CHECK-LABEL: @vmla_cmp
+// CHECK-SAME: %[[LHS:[a-zA-Z0-9$._-]+]]
```

Comment:
meta comment: these full regexes aren't needed and just add noise - `.+` is enough to capture everything until the space and `%[[LHS:.+]]` is both easier to write and easier to read

can you find/replace?

Reply:
I don't think `.+` stops on spaces, so that's not quite right. We've already got a fair number of examples of this method of capturing args. But there are a few other ways. My current principle is to try to use `.+` where I can (which always works for capturing SSA values) and otherwise use the full regex. I've seen some various other creative ways to do it (Stella did something with explicitly capturing not space, and I think sometimes people write types). I suggest we come up with a consistent way we want to capture args and then we can do a big cleanup, but let's discuss it like we did for capturing sigils or not.

Reply:
SGTM
I haven't hit a case where .+ hasn't worked, and I'd encourage trying to build matches that it works for. And if lit wasn't lit then we could have nice named regexes matching the grammar keywords like `literal` that would be much easier to grok.

`\S` may work (depending on which regex engine is used by lit)
`[[LHS:\S+]]` is still pretty nice and concise

Reply:
`\S` doesn't seem to work in checkfile, however `%[[LHS:[^:[:space:]]+]]` does. Here's a side-by-side.
```
%[[LHS:[^:[:space:]]+]]
%[[LHS:[a-zA-Z0-9$._-]+]]
```
Thoughts? I think I would be able to figure out what `^:[:space:]` does if I encountered it without already knowing, but it's impossible to Google that and get anything not related to NASA 🤦

Reply:
I think the problem is that `.+` works in cases where there are characters in the same check line to signal the `.+` when to stop, but if a check is split up over multiple lines then it stops working. E.g.:
```
// CHECK: %[[EPS:.+]] = xla_hlo.constant dense<1.001000e-05> : tensor<f32>
```
would work, but
```
// CHECK:      %[[EPS:.+]]
// CHECK-SAME: = xla_hlo.constant dense<1.001000e-05> : tensor<f32>
```
fails because the `.+` blindly eats everything on the line.

Reply:
I've opened a separate issue for this discussion (https://github.com/google/iree/issues/1795). Let's merge this PR.